version: '3.8'

services:
  app:
    image: ghcr.io/juanfeliperestrepobuitrago/herzonteis/herzon_ecommerce:latest
    container_name: herzon_app
    environment:
      DATABASE_URL: jdbc:postgresql://postgres:5432/herzon_db
      DATABASE_USERNAME: root
      DATABASE_PASSWORD: 123
    restart: always
    depends_on:
      - postgres
    networks:
      - app_network

  postgres:
    image: postgres:16
    container_name: herzon_postgres
    environment:
      POSTGRES_USER: root
      POSTGRES_PASSWORD: 123
      POSTGRES_DB: herzon_db
      TZ: "America/Bogota"
    restart: always
    volumes:
      - postgres_data:/var/lib/postgresql/data
    networks:
      - app_network

  pgadmin:
    image: dpage/pgadmin4:latest
    container_name: herzon_pgadmin
    environment:
      PGADMIN_DEFAULT_EMAIL: admin@herzon.com
      PGADMIN_DEFAULT_PASSWORD: 123
    restart: always
    networks:
      - app_network

  nginx-proxy-manager:
    image: jc21/nginx-proxy-manager:latest
    container_name: nginx-proxy-manager
    restart: always
    ports:
      - "80:80"
      - "81:81"
      - "443:443"
    environment:
      DB_SQLITE_FILE: "/data/database.sqlite"
    volumes:
      - ./.nginx-proxy-manager/data:/data
      - ./.nginx-proxy-manager/letsencrypt:/etc/letsencrypt
    networks:
      - app_network
    healthcheck:
      test: ["CMD", "curl", "-f", "http://127.0.0.1:81"]
      interval: 1m
      timeout: 10s
      retries: 5

networks:
  app_network:
    driver: bridge

volumes:
  postgres_data: